apps/backend/src/modules/users/user.repository.ts

```diff
@@ -47,10 +49,22 @@ class UserRepository implements Repository {
 			.query()
 			.findById(id)
 			.whereNull("deletedAt")
-			.returning("*")
-			.execute();
+			.withGraphFetched("groups.permissions")
+			.modifyGraph("groups", (builder) => {
+				builder.select("user_groups.id", "name");
+			})
+			.modifyGraph("groups.permissions", (builder) => {
+				builder.select("permissions.id", "name", "key");
+			})
+			.castTo<
```

Comment:
I've just checked it locally and we don't need this castTo at all, we can just add .execute() at the end

Reply:
Oh, thanks a lot for the help, I don't know why I didn't try it right away